### PR TITLE
Honda: Odyssey 2021-25

### PR DESCRIPTION
**Pull request by csouers**
**Checklist**

- [ ] (when out of dashcam) added entry to CAR in selfdrive/car/*/values.py and ran `selfdrive/opcar/docs.py` to generate new docs
- [ ] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [ ] route with openpilot:
- [ ] route with stock system: might have this from before
- [x] car harness used (if comma doesn't sell it, put N/A): `Bosch A`

Prereqs:
- https://github.com/commaai/opendbc/pull/1330

### CODE REVIEW

File: selfdrive/car/car_specific.py [hunk 1]
```diff
@@ -88,6 +88,23 @@ def update(self, CS: CarStateBase, CS_prev: car.CarState, CC_prev: car.CarContro
       if self.CP.minEnableSpeed > 0 and CS.out.vEgo < 0.001:
         events.add(EventName.manualRestart)
 
+      # Low speed steer alert logic; for cars with steer cut off above 6 m/s
+      # Show the alert earlier when slowing. Must first exceed a greater speed while engaged.
+      if CS.out.vEgo > (self.CP.minSteerSpeed + 3.5):
```
Comment (csouers) [inline selfdrive/car/car_specific.py:93]: I chose this policy as users have reported that continuing to see the alert just after the car has began to steer can be confusing.

File: selfdrive/car/car_specific.py [hunk 2]
```diff
@@ -88,6 +88,23 @@ def update(self, CS: CarStateBase, CS_prev: car.CarState, CC_prev: car.CarContro
...
+        CS.min_steer_alert_speed = (self.CP.minSteerSpeed + 1.5)
+      elif CS.out.vEgo <= self.CP.minSteerSpeed or not CC_prev.enabled:
+        CS.min_steer_alert_speed = self.CP.minSteerSpeed
+      self.low_speed_alert = (0 < CS.out.vEgo <= CS.min_steer_alert_speed) and self.CP.minSteerSpeed > 6.0
+      if self.low_speed_alert:
+        events.add(EventName.belowSteerSpeed)
+
+      # Some cars forcibly disengage steering depending on vehicle conditions (i.e. some Odyssey Bosch & late model Acura RDX 3G).
```
Comment (csouers) [inline selfdrive/car/car_specific.py:101]: could split this off for ease of merging

**Comment by jyoung8607** [comment]
Just to set expectations, @sshane is in the middle of a big cleanup and refactor of car events. We'll still be able to accomplish what you need, but this particular PR will be on hold for a short time until that refactor merges.

File: selfdrive/car/car_specific.py [hunk 3]
```diff
@@ -86,6 +86,23 @@ def update(self, CS: car.CarState, CS_prev: car.CarState, CC: car.CarControl):
...
 
+      # Low speed steer alert logic; for cars with steer cut off above 6 m/s
+      # Show the alert earlier when slowing. Must first exceed a greater speed while engaged.
+      if CS.out.vEgo > (self.CP.minSteerSpeed + 3.5):
+        CS.min_steer_alert_speed = (self.CP.minSteerSpeed + 1.5)
+      elif not CC.latActive:
+        CS.min_steer_alert_speed = self.CP.minSteerSpeed
+      self.low_speed_alert = (0 < CS.out.vEgo <= CS.min_steer_alert_speed) and self.CP.minSteerSpeed > 6.0
+      if self.low_speed_alert:
+        events.add(EventName.belowSteerSpeed)
```
Comment (sshane) [inline selfdrive/car/car_specific.py:97]: Can you move all this into the interface and use lowSpeedAlert?

Comment (jyoung8607) [reply selfdrive/car/car_specific.py:97]: FYI @csouers this is the refactor I was talking about above. There's a good example to follow in #33712, converting Mazda to use the new common mechanism.

Comment (csouers) [reply selfdrive/car/car_specific.py]: Nice. Done.

As for the steer_off_cnt, i need a way to get the prev latActive or similar inside of carstate. Not seeing a way as things are.

Comment (csouers) [reply selfdrive/car/car_specific.py]: oh wait! i see a loopback can parser is a thing. neat. will try that

**Pull request closed without merging**